src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java

```diff
@@ -615,6 +619,24 @@ private void handleEditorChangeReceived(JSONObject changeMessage) {
         }
 
         /**
+         * Remove a change from the connected UI.
+         */
+        private void handleEditorBindingsCleared(JSONObject clearMessage) {
+            try {
+                final JSONObject payload = clearMessage.getJSONObject("payload");
+                final JSONArray actions = payload.getJSONArray("actions");
+                for (int i = 0; i < actions.length(); i++) {
+                    final String changeId = actions.getString(i);
+                    mEditorChanges.remove(changeId);
```

Comment:
Take care of it

-----Original Message-----
From: "yinfeiru" <notifications@github.com>
Sent: ‎25/‎03/‎2015 08:09 PM
To: "mixpanel/mixpanel-android" <mixpanel-android@noreply.github.com>
Subject: Re: [mixpanel-android] Basic undo. We'll need to revisit text sizefor this to work (#193)

In src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java:
> @@ -615,6 +619,24 @@ private void handleEditorChangeReceived(JSONObject changeMessage) {
>          }
>  
>          /**
> +         * Remove a change from the connected UI.
> +         */
> +        private void handleEditorBindingsCleared(JSONObject clearMessage) {
> +            try {
> +                final JSONObject payload = clearMessage.getJSONObject("payload");
> +                final JSONArray actions = payload.getJSONArray("actions");
> +                for (int i = 0; i < actions.length(); i++) {
> +                    final String changeId = actions.getString(i);
> +                    mEditorChanges.remove(changeId);
I'm not that familiar with JAVA ArrayList, but what if changeId doesn't exist?
—
Reply to this email directly or view it on GitHub.